lib/psulib_traject/workers/hourly_indexer.rb

```diff
@@ -4,14 +4,56 @@ module PsulibTraject
   module Workers
     class HourlyIndexer < Base
       def perform
-        target = Dir.glob("#{ConfigSettings.symphony_data_path}/#{ConfigSettings.symphony_hourlies_subdir}/**/*.m*rc")
+        perform_indexes
+        perform_deletes
+      end
+
+      def hourlies_directory
+        @hourlies_directory ||= Pathname
+          .new(ConfigSettings.symphony_data_path)
+          .join(ConfigSettings.symphony_hourlies_subdir)
+      end
+
+      def perform_deletes
+        target_deletes = hourlies_directory
+          .children
+          .select { |filename| filename.basename.fnmatch?('*_deletes_*.txt') }
+
+        processed_deletes = redis.keys('hr:*').map { |e| e.gsub('hr:', '') }
+        files_to_process = target_deletes.reject { |filename| processed_deletes.include?(filename.to_s) }
+
+        indexer.logger.info "Found #{files_to_process.length} files to process for deletes"
+
+        files_to_process.each do |file_name|
+          File
+            .read(file_name)
+            .split("\n")
+            .map { |id| delete(id) }
+          redis.set("hr:#{file_name}", true)
+        end
+      end
+
+      def delete(id)
+        indexer.writer.delete(id)
+        indexer.logger.info "Deleted ID #{id}"
+      end
+
+      def perform_indexes
+        target_files = hourlies_directory
+          .children
+          .select { |filename| filename.extname.match?(/ma?rc/) }
+
         indexed_files = redis.keys('hr:*').map { |e| e.gsub('hr:', '') }
-        files_to_index = target - indexed_files
+        files_to_index = target_files.reject { |filename| indexed_files.include?(filename.to_s) }
+
+        indexer.logger.info "Found #{files_to_index.length} files to index"
+        array_of_files = files_to_index.map { |file| File.new(file) }
+
+        indexer.process array_of_files
```

Comment:
Did you try using `files_to_index` here? It should work because Pathname wraps all the functions of File.

Reply:
I missed this, I just tried it, and marc4j_reader throws the following 

       undefined method `to_inputstream' for <Pathname:./spec/fixtures/hourlies/date_008.marc>:Pathname

Reply:
Ah ok, I think this might be a weird thing with jruby.